### PR TITLE
Bubble upstream the latest changes on libp2p

### PR DESCRIPTION
**Pull request by diasdavid**
- [x] Add new dial calls to libp2p
- [x] Make more robust general connectivity tests
- [x] Make turbulence tests

### CODE REVIEW

**Comment by diasdavid** [comment]
Node.js tests work like a charm, however, the browser tests fails on the dial (no available transports), this typically means that the listener didn't start or that the dialer doesn't have the required transport.

@haadcode mentioned that libp2p-ipfs-browser is not being injected properly and after a few attempts playing with .aegir.js, I didn't manage to have a success confirming this. @dignifiedquire can you shine some light of Webpack goodness into this PR? :)

**Comment by diasdavid** [comment]
@dignifiedquire nevermind, all good now :)

File: package.json [hunk 1]
```diff
@@ -37,14 +38,15 @@
...
-    "aegir": "^3.1.0",
+    "aegir": "^3.2.0",
     "buffer-loader": "0.0.1",
     "chai": "^3.5.0",
     "expose-loader": "^0.7.1",
     "form-data": "^1.0.0-rc3",
     "gulp": "^3.9.1",
     "idb-plus-blob-store": "^1.1.2",
     "interface-ipfs-core": "^0.1.5",
+    "left-pad": "^1.1.0",
```
Comment (dignifiedquire) [inline package.json]: are you sure you want to use this module :D

File: test/core/node-only/test-libp2p.js [hunk 2]
```diff
@@ -0,0 +1 @@
+'use strict'
```
Comment (dignifiedquire) [inline test/core/node-only/test-libp2p.js]: why this empty file?

File: test/core/node-only/test-swarm.js [hunk 3]
```diff
@@ -1,77 +1,132 @@
 /* eslint-env mocha */
+
```
Comment (dignifiedquire) [inline test/core/node-only/test-swarm.js]: not needed

File: test/core/node-only/test-swarm.js [hunk 4]
```diff
@@ -1,77 +1,132 @@
...
+              expect(Object.keys(res)).to.have.length(0)
+              cb()
+            })
+          }
+        ], done)
+      }
     })
   })
+
+  it.skip('stop', (done) => {
```
Comment (dignifiedquire) [inline test/core/node-only/test-swarm.js]: why is this skipped?

Comment (diasdavid) [inline test/core/node-only/test-swarm.js]: it shouldn't. Fixed

File: test/core/node-only/test-swarm.js (hunk 3 above)
Comment (diasdavid) [inline test/core/node-only/test-swarm.js]: deleted

File: test/core/node-only/test-libp2p.js (hunk 2 above)
Comment (diasdavid) [inline test/core/node-only/test-libp2p.js]: deleted

File: package.json (hunk 1 above)
Comment (diasdavid) [inline package.json]: ahaha I was waiting for someone to catch this :D

**Comment by diasdavid** [comment]
@dignifiedquire any other comments? Good to merge and continue tackling on more tests in more PR (this was about making libp2p robustness tests, but then it became more about making libp2p-ipfs more powerful and fix a bunch of bugs along the way)

**Comment by dignifiedquire** [comment]
go for it, I think we ahoul encapaulate even more in libp2p that we are doing currently in js-ipfs but that can come later

**Pull request merged**